### PR TITLE
sunburst chart added solved issue #700

### PR DESCRIPTION
**Pull request by ath34-tech**
I simply add sunburst chart dependency from taking inspiration from https://plotly.com/javascript/sunburst-charts/ and looking at pie() function. this function is inspired by https://plotly.com/javascript/sunburst-charts/#basic-sunburst-chart
I will do further changes and make it more advance in future.

## Description
I added a simple function to achieve this:
```
    @pytorch_wrap
    def sunburst(self, X,parents,labels, win=None, env=None, opts=None):
        opts = {} if opts is None else opts
        _title2str(opts)
        _assert_opts(opts)
        
        font_size=opts.get("size")
        font_color=opts.get("font_color")
        opacity=opts.get("opacity")
        line_width=opts.get("marker_width")

        X = np.squeeze(X)
        assert X.ndim == 1, 'X should be one-dimensional'
        assert np.all(np.greater_equal(X, 0)), \
            'X cannot contain negative values'


        data = [{
            'values': X.tolist(),
            'labels': labels.tolist(),
            "parents":parents.tolist(),
            "outsidetextfont":{"size":font_size,"color":font_color},
            "leaf":{"opacity":opacity},
            "marker":{"line":{"width":line_width}},
            'type': 'sunburst',
        }]
        return self._send({
            'data': data,
            'win': win,
            'eid': env,
            'layout': _opts2layout(opts),
            'opts': opts,
        })

```
## Motivation and Context
It solve issue #700 

## How Has This Been Tested?
Code snippet(inspired from demo.py's pie code):

```
from visdom import Visdom
import numpy as np
viz=Visdom()
X = np.asarray( [10, 14, 12, 10, 2, 6, 6, 4, 4])
viz.sunburst(X=X,labels=np.asarray(["Eve", "Cain", "Seth", "Enos", "Noam", "Abel", "Awan", "Enoch", "Azura"]),parents=np.asarray(["", "Eve", "Eve", "Seth", "Seth", "Eve", "Eve", "Awan", "Eve" ]))

```

## Screenshots (Output):
visdom

I will make this function more advance in future, currently it is fully inspired by:https://plotly.com/javascript/sunburst-charts/#basic-sunburst-chart

### CODE REVIEW

File: py/visdom/__init__.py [hunk 1]
```diff
@@ -2128,6 +2128,42 @@ def stem(self, X, Y=None, win=None, env=None, opts=None):
...
+        _assert_opts(opts)
+        
+        font_size=opts.get("size")
+        font_color=opts.get("font_color")
+        opacity=opts.get("opacity")
+        line_width=opts.get("marker_width")
+
+        X = np.squeeze(X)
+        assert X.ndim == 1, 'X should be one-dimensional'
+        assert np.all(np.greater_equal(X, 0)), \
```
Comment (JackUrb) [inline py/visdom/__init__.py]: The values for `'values'` (which come from `X` in your implementation) aren't required based on the sunburst, and is there any reason you're forcing them to be positive?

File: py/visdom/__init__.py [hunk 2]
```diff
@@ -2128,6 +2128,42 @@ def stem(self, X, Y=None, win=None, env=None, opts=None):
...
+            'X cannot contain negative values'
+
+
+        data = [{
+            'values': X.tolist(),
```
Comment (JackUrb) [inline py/visdom/__init__.py]: Values can be left out if this is not provided.

File: py/visdom/__init__.py [hunk 3]
```diff
@@ -2128,6 +2128,42 @@ def stem(self, X, Y=None, win=None, env=None, opts=None):
 
         return self.scatter(X=data, Y=labels, opts=opts, win=win, env=env)
 
+
+    @pytorch_wrap
+    def sunburst(self, X,parents,labels, win=None, env=None, opts=None):
```
Comment (JackUrb) [inline py/visdom/__init__.py]: The function signature should likely be:
```
def sunburst(self, labels, parents, values=None, win=None, env=None, opts=None):
```

File: py/visdom/__init__.py [hunk 4]
```diff
@@ -2128,6 +2128,42 @@ def stem(self, X, Y=None, win=None, env=None, opts=None):
...
+        _title2str(opts)
+        _assert_opts(opts)
+        
+        font_size=opts.get("size")
+        font_color=opts.get("font_color")
+        opacity=opts.get("opacity")
+        line_width=opts.get("marker_width")
+
+        X = np.squeeze(X)
+        assert X.ndim == 1, 'X should be one-dimensional'
```
Comment (JackUrb) [inline py/visdom/__init__.py]: Outside of this, reasonable assertions would be that `labels` and `parents` are the same size, and another for that `values` is as well (if `values` is provided).

**Review by JackUrb — CHANGES_REQUESTED** [review]
Thanks for the contribution here @ath34-tech. This is generally in the right direction, though needs a little bit of modification to be ready.

After the noted changes, it would also be useful to include documentation for the function (both in a doc-block in `__init__.py` as all the other functions have as well as in the `README`). You can also add your example into the core `demo.py` file.

**Review by ath34-tech — COMMENTED** [review]
Hey there, i done as you said, please check!

**Comment by ath34-tech** [comment]
I added neccessary assertions, and checked them by running they are running properly here and it look all ok

File: py/visdom/__init__.py [hunk 5]
```diff
@@ -2128,6 +2128,52 @@ def stem(self, X, Y=None, win=None, env=None, opts=None):
...
+        
+        else:
+            data = [{
+                'labels': labels.tolist(),
+                "parents":parents.tolist(),
+                "outsidetextfont":{"size":font_size,"color":font_color},
+                "leaf":{"opacity":opacity},
+                "marker":{"line":{"width":line_width}},
+                'type': 'sunburst',
+            }]
```
Comment (JackUrb) [inline py/visdom/__init__.py]: This is a particularly ineffective way to deal with this, as if you ever want to change the remains of the data packet you'd need to change in both locations.

You can do something like:
```
data_dict = {...}
if values is not None:
    ...
    data_dict['values'] = ...

data = [data_dict]
```

**Review by JackUrb — CHANGES_REQUESTED** [review]
Nit on use of duplicate code, still waiting on docs + demo inclusion.

**Comment by ath34-tech** [comment]
Yes thank for suggestion!
and for docs and demo I will provide them soon. till tomorrow or day after tomorrow.

**Review by ath34-tech — COMMENTED** [review]
I edited the file again as you suggested, merge this branch if it is ok to do so without any demo and readme tutorial. I will provide them tomorrow.
Thank you!

**Comment by JackUrb** [comment]
I prefer to include docs for code in the same PR that the functionality is added.

**Comment by ath34-tech** [comment]
Hey @JackUrb i added docs.

**Review by ath34-tech — COMMENTED** [review]
I added docs,please check them.

**Review by JackUrb — APPROVED** [review]
Thanks for adding this functionality!

**Pull request merged**